### PR TITLE
add error handling to Curl module

### PR DESCRIPTION
**Pull request by psahabu**
Fixes the issues raised in #7570 by marking `ioerror()` calls with `try` and making the calling methods throw.

- [x] passes linux64+flat testing

### CODE REVIEW

**Comment by bradcray** [comment]
I'm leaving this one to @mppf since he's more familiar with both error-handling and the curl module.

**Comment by mppf** [comment]
This patch looks fine to me. @psahabu - the worrying thing is that this issue did not cause test failures. That means that these functions are not tested nightly. Why not? Do we have tests for them at all? Do they pass now? (If we have them, we need to run them nighty. If we don't have them, we need to add them).

**Comment by bradcray** [comment]
I agree with Michael's point in the following respect: If a PR is fixing a previously unknown bug, it should also include a new test that would've failed before and passes now to lock in the fix.

W.r.t. the broader question of what kind of curl testing we already have, I don't recall.  But this wouldn't be the first instance of a library/feature that has insufficient testing... (not to excuse it, just to say that many of us are guilty of this).

**Comment by psahabu** [comment]
@mppf it does not appear that we have Curl tests. I've added the repro in #7570 as a test.

Is it possible that there aren't Curl tests because they might be inconsistent from machine to machine?

**Comment by mppf** [comment]
@psahabu - I suspect it was simply an oversight. PR #89 added Curl support and that PR message refers to an apparently uncommitted set of unit tests:   https://github.com/tzakian/chapel-unit-tests/tree/curl

Could you look at migrating those tests in to the Chapel repo? You'll need a SKIPIF to skip them if curl support is not enabled. (And we'll need to enable curl in some nightly testing config).

**Comment by psahabu** [comment]
I'm taking a look at the tests now.

**Comment by mppf** [comment]
@psahabu - perhaps we should merge this PR, but continue to look at testing Curl nightly as a separate effort?

File: test/modules/packages/Curl/setopt.chpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use Curl;
+
+proc main() {
+  const uniprotid = "P07204_TRBM_HUMAN";
+  const uniproturl = "http://www.uniprot.org/uniprot/" + uniprotid + ".fasta";
```
Comment (mppf) [inline test/modules/packages/Curl/setopt.chpl]: Seems to me that we should probably create testing traffic against our own website

File: test/modules/packages/Curl/setopt.good [hunk 2]
```diff
@@ -0,0 +1 @@
+internal error: No Curl Support
```
Comment (mppf) [inline test/modules/packages/Curl/setopt.good]: Is this a compilation failure or a runtime failure? I'd expect the .good file to have the normal output and then the .skipif file to skip the test if curl is not available.

**Comment by psahabu** [comment]
I'll split out the setopt tests and merge this PR, then open another PR when the rehabilitated Curl tests are ready.

**Comment by psahabu** [comment]
Supplanted by #7713.

**Pull request closed without merging**